sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRegistryAsyncClient.java

```diff
@@ -128,55 +91,71 @@
 
     Mono<PagedResponse<String>> listRepositoriesNextSinglePageAsync(String nextLink, Context context) {
         try {
-            Mono<PagedResponse<String>> pagedResponseMono = this.serviceClient.listRepositoriesNextSinglePageAsync(nextLink, context);
-            return pagedResponseMono.map(res -> getPagedResponseWithContinuationToken(res));
+            Mono<PagedResponse<String>> pagedResponseMono = this.registriesImplClient.getRepositoriesNextSinglePageAsync(nextLink, context);
+            return pagedResponseMono.map(res -> Utils.getPagedResponseWithContinuationToken(res));
         } catch (RuntimeException e) {
             return monoError(logger, e);
         }
     }
 
-
     /**
      * Delete the repository identified by `name`.
      *
      * @param name Name of the image (including the namespace).
      * @return deleted repository.
-     * @throws IllegalArgumentException thrown if parameters fail the validation.
-     * @throws AcrErrorsException thrown if the request is rejected by server.
-     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws ResourceNotFoundException thrown if the given name was not found.
+     * @throws NullPointerException thrown if the name is null.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<Response<DeleteRepositoryResult>> deleteRepositoryWithResponse(String name) {
         return withContext(context -> deleteRepositoryWithResponse(name, context));
     }
 
     Mono<Response<DeleteRepositoryResult>> deleteRepositoryWithResponse(String name, Context context) {
-        if (CoreUtils.isNullOrEmpty(name)) {
-            return Mono.error(logger.logExceptionAsError(new NullPointerException("repository name can't be null or empty.")));
-        }
+        try {
+            if (name == null) {
+                return monoError(logger, new NullPointerException("'name' cannot be null."));
+            }
+
+            if (context == null) {
+                return monoError(logger, new NullPointerException("'context' cannot be null."));
+            }
 
-        return this.serviceClient.deleteRepositoryWithResponseAsync(name, context);
+            return this.registriesImplClient.deleteRepositoryWithResponseAsync(name, context)
+                .map(res -> Utils.mapResponse(res, Utils::mapDeleteRepositoryResult))
+                .onErrorMap(Utils::mapException);
```

Comment:
This can easily be part of the proxy and all generic error codes can\should be handle by Rest proxy unless the library requires special handling.

401, 400, 503, 409 etc.